app/lib/redis_configuration.rb

```diff
@@ -39,12 +39,38 @@ def namespace
   end
 
   def url
-    ENV['REDIS_URL']
+    if ENV['REDIS_SENTINEL']
+      m = ENV.fetch('REDIS_SENTINEL_MASTER', 'mymaster')
+      p = ENV.fetch('REDIS_PASSWORD', '')
+      "redis://:#{p}@#{m}"
+    else
+      ENV['REDIS_URL']
+    end
+  end
+
+  def sentinels
+    return unless ENV['REDIS_SENTINEL']
+
+    unless ENV['REDIS_SENTINEL'].include? ','
+      ips = Resolv.getaddresses(ENV['REDIS_SENTINEL'])
```

Comment:
You might hit some weirdness here if using something like Tailscale on your infrastructure, which manages connectivity.. (you'll need to explicitly start mastodon after tailscale is up, not just after networking is up.

Reply:
Something else: DNS might update itself to dynamically walk IPs to new failover, and this would not be updating those IPs

i dont know if the redis client will internally keep resolving these hostnames, but this might be a concern

Reply:
It is a little unclear from the docs whether it will resolve the addresses. My assumption is that yes but needs testing, and I was going off what the docs indicated. You are right though - since we do this at init it would be an issue. Digging into this.

Reply:
All the docs only ever show IP addresses used for sentinels, but I would assume as well that the client will work with hostnames just as well.

If you are able to verify that it does, I too believe we should not this extra work here.